openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -11,9 +11,10 @@
 def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
     """
     Catches the signal that a course has been published in Studio and
-    invalidates the corresponding CourseOverview cache entry if one exists.
+    updates the corresponding CourseOverview cache entry.
     """
     CourseOverview.objects.filter(id=course_key).delete()
+    CourseOverview.load_from_module_store(course_key)
```

Comment:
@doctoryes I have this same concern and it keeps me awake.  We are adding more and more features now that do relatively expensive work at every course publish!

Having helped implement the changes in the course-publish-signal PR and being aware of the technical debt we have in not-updating Studio to have an explicit Publish button for all course-module-type changes, I know that these signals can be fired more often than we really want.  I had raised this concern in the course-publish-signal PR and wondered whether we could:
1. batch process the firing of the signal, or
2. have more fine-grained signals (or more detail in the signal so the subscribers know what type of change was made).

It's exactly because of these concerns that we initially chose to compute the CourseOverview object lazily, rather than at each publish.

But for now, I am trusting @ormsbee's instincts for this.  It may be that we'll need to address the course publish signal soon for the platform overall.